src/lib.rs

```diff
@@ -239,21 +239,66 @@ impl<T> Status<T> {
 }
 
 /// Parser configuration.
-#[derive(Clone, Debug, Default)]
+#[derive(Clone)]
 pub struct ParserConfig {
     allow_spaces_after_header_name_in_responses: bool,
     allow_obsolete_multiline_headers_in_responses: bool,
     allow_multiple_spaces_in_request_line_delimiters: bool,
     allow_multiple_spaces_in_response_status_delimiters: bool,
     ignore_invalid_headers_in_responses: bool,
+
+    version_parser: fn(&mut Bytes) -> Result<u8>,
+}
+
+impl fmt::Debug for ParserConfig {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("ParserConfig")
+         .field("allow_spaces_after_header_name_in_responses", &self.allow_spaces_after_header_name_in_responses)
+         .field("allow_obsolete_multiline_headers_in_responses", &self.allow_obsolete_multiline_headers_in_responses)
+         .field("allow_multiple_spaces_in_request_line_delimiters", &self.allow_multiple_spaces_in_request_line_delimiters)
+         .field("allow_multiple_spaces_in_response_status_delimiters", &self.ignore_invalid_headers_in_responses)
+         .field("ignore_invalid_headers_in_responses", &self.ignore_invalid_headers_in_responses)
+         .finish()
+    }
+}
+
+impl Default for ParserConfig {
+    fn default() -> Self {
+        Self::const_default()
+    }
 }
 
 impl ParserConfig {
+    /// Returns default configuration. Same as `Default::default()` but can be used in const
+    /// context.
+    pub const fn const_default() -> Self {
+        Self {
+            allow_spaces_after_header_name_in_responses: false,
+            allow_obsolete_multiline_headers_in_responses: false,
+            allow_multiple_spaces_in_request_line_delimiters: false,
+            allow_multiple_spaces_in_response_status_delimiters: false,
+            ignore_invalid_headers_in_responses: false,
+            version_parser: HTTP_VERSION_PARSER,
+        }
+    }
+
+    /// Sets first-line parsing to SIP
+    pub const fn set_sip_protocol_parser(mut self) -> Self {
+        self.version_parser = SIP_VERSION_PARSER;
+        self
```

Comment:
I would love to see this as something like the following:
```Rust
pub const fn set_protocol_parser(mut self, protocol: Protocol) -> Self {
    self.version_parser = match protocol {
        Protocol::Http => HTTP_VERSION_PARSER,
        Protocol::Sip => SIP_VERSION_PARSER,
    };
    self
}
```

However, that gives the following compilation error:
```
error[E0658]: mutable references are not allowed in constant functions
   --> src/lib.rs:287:31
    |
287 |           self.version_parser = match protocol {
    |  _______________________________^
288 | |             Protocol::Http => HTTP_VERSION_PARSER,
289 | |             Protocol::Sip => SIP_VERSION_PARSER,
290 | |         };
    | |_________^
    |
    = note: see issue #57349 <https://github.com/rust-lang/rust/issues/57349> for more information
```
which is weird :|

Reply:
@gootorov Similar code works fine [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=24706ed9909d39bfeb4f5cc830a57a81). What is the difference?

Reply:
I mean, is it only the attribute type, or the `Protocol` enum too?